src/status_im2/contexts/add_new_contact/events.cljs

```diff
@@ -86,15 +84,35 @@
                                :ens-name ens-name}
                               (fx-callbacks id ens-name))})))
 
+(rf/defn build-contact
+  {:events [:contacts/build-contact]}
+  [_ pubkey open-profile-modal?]
+  {:json-rpc/call [{:method     "wakuext_buildContact"
+                    :params     [pubkey]
+                    :on-success #(rf/dispatch [:contacts/contact-built
+                                               pubkey
+                                               open-profile-modal?
+                                               (data-store.contacts/<-rpc %)])}]})
+
+(rf/defn contact-built
+  {:events [:contacts/contact-built]}
+  [{:keys [db] :as cofx} pubkey open-profile-modal? contact]
+  (rf/merge cofx
```

Comment:
just a bit shorted as `rf/merge` is not really needed here
```Clojure
(merge {:db (assoc-in db [:contacts/contacts pubkey] contact)}
       (when open-profile-modal?
        {:dispatch [:open-modal :profile]}))
 ```
 or maybe
 ```clojure
 {:db (assoc-in db [:contacts/contacts pubkey] contact)
  :dispatch (when open-profile-modal? [:open-modal :profile])}
 ```
 I also saw some issues when `:dispatch` has to be merged (because there is no proper way to merge it without making it `:dispatch-n`) so I would probably go with `:dispatch-n` right away
  ```clojure
 {:db (assoc-in db [:contacts/contacts pubkey] contact)
   :dispatch-n (when open-profile-modal? [[:open-modal :profile]])}
 ```
 `¯\_(ツ)_/¯`

Reply:
In one of the previous reviews I was told to never ever use `dispatch-n` for some reason :D 
But I updated with the first version, it definitely makes sense for me.